### PR TITLE
HOUSNAV-200: Updated Application State

### PR DESCRIPTION
**Pull request by nicholaslyon-dt**
[HOUSNAV-200](https://hous-bssb.atlassian.net/browse/HOUSNAV-200)

## Overview & Purpose
Update the application state to account for a walkthrough that has multiple sets of walkthrough questions and results.

## Summary of Changes
- Updated walkthrough URLS to use a query parameter
- Query parameter(s) are used to build the full walkthrough data
- Updated application store code to account for the possibility of multiple walkthroughs with different sets of questions and results

## Side Effects
All of them?

## Testing
- Make sure the current single dwelling walkthroughs work
- Make sure different configurations of multi dwelling walkthroughs allow the user to get to a result

## Notes & Resources
- The step tracker and results updates are captured in separate tickets

## Checklist
- [x] I have written or updated vitests for this work
- [x] I have reviewed the [accessibility guidelines](https://digital.gov.bc.ca/wcag/home/) relevant to this work
- [x] I have reviewed this work with at least one screen reader (when applicable)
- [x] I have added/updated any related documentation

### CODE REVIEW

**Comment by jbook24** [comment]
Design 
image


Current 
image


- Are we thinking the step tracker will update to reflect the current design from Figma? 
- Currently if you select "all" it only shows the new section after completing the prior

File: apps/web/components/question/Question.test.tsx [hunk 1]
```diff
@@ -94,7 +103,7 @@ describe("Question", () => {
   // check question with code reference
   it("QuestionMultiChoice: renders with code reference", () => {
     // get data
-    const walkthroughData = useWalkthroughTestData();
+    const walkthroughData = useWalkthroughTestData999();
```
Comment (jbook24) [inline apps/web/components/question/Question.test.tsx:106]: As we prepare for significant uptick in use cases do we feel this approach is scalable?

File: apps/web/stores/AnswerStore.ts [hunk 2]
```diff
@@ -87,19 +113,26 @@ export class AnswerStore {
   setAnswerValueOnChange = (value: AnswerTypes, questionId: string) => {
     this.eraseAnswersAfterCurrentItem();
 
-    this.rootStore.navigationStore.farthestItemId = questionId;
+    this.rootStore.navigationStore.setFarthestIds(questionId);
 
-    this.answers[questionId] = value;
+    this.setAnswerValue(
+      this.rootStore.navigationStore.currentWalkthroughId,
```
Comment (jbook24) [inline apps/web/stores/AnswerStore.ts:119]: Is it possible for this "id" to ever be out of sync?

File: apps/web/stores/NavigationStore.ts [hunk 3]
```diff
@@ -50,18 +64,35 @@ export class NavigationStore {
     return "";
   }
 
+  setFarthestIds = (itemId: string, walkthroughId?: string) => {
```
Comment (jbook24) [inline apps/web/stores/NavigationStore.ts:67]: What does this mean?

File: apps/web/stores/NavigationStore.ts [hunk 4]
```diff
@@ -90,44 +121,54 @@ export class NavigationStore {
 
   /*
    * Back button is disabled if:
-   * - current question is the first question
+   * - current question is the first question in the history
    */
   get backButtonIsDisabled() {
```
Comment (jbook24) [inline apps/web/stores/NavigationStore.ts:126]: Should this "back button" be enabled to allow for users to update the code selections they want to review? 

image

File: packages/data/json/building-types/multi-dwelling/wt-multi-dwelling-9.10.14.json
Comment (jbook24) [inline packages/data/json/building-types/multi-dwelling/wt-multi-dwelling-9.10.14.json:1]: This will be updated when the SME shares the updated version?

**Comment by nicholaslyon-dt** [comment]
> Design image
> 
> Current image
> 
> * Are we thinking the step tracker will update to reflect the current design from Figma?
> * Currently if you select "all" it only shows the new section after completing the prior

@jbook24 Yes, I was planning for all updates related to the step tracker and results screen will be handled in the [step tracker](https://hous-bssb.atlassian.net/browse/HOUSNAV-199) and [results](https://hous-bssb.atlassian.net/browse/HOUSNAV-191) tickets.

File: apps/web/components/question/Question.test.tsx (hunk 1 above)
Comment (nicholaslyon-dt) [reply apps/web/components/question/Question.test.tsx:106]: Yeah, this update was just to be more explicit here. This is just so the test reads with exactly what type of test data it is using. I don't see the need for the tests to start using all the data from different types of walkthroughs.

File: apps/web/stores/AnswerStore.ts (hunk 2 above)
Comment (nicholaslyon-dt) [reply apps/web/stores/AnswerStore.ts:119]: I don't think so. This only happens within the context of 1 question, so the current walkthrough id wouldn't change while a user is interacting with a question. And all state is updated before any rendering happens between questions.

File: apps/web/stores/NavigationStore.ts (hunk 3 above)
Comment (nicholaslyon-dt) [reply apps/web/stores/NavigationStore.ts:67]: It's just a helper function to update both of the stored values for the farthest question and farthest walkthrough the user has currently been to.

File: apps/web/stores/NavigationStore.ts (hunk 4 above)
Comment (nicholaslyon-dt) [reply apps/web/stores/NavigationStore.ts:126]: No, this is what the breadcrumbs would be for. That back only ever operates within a walkthrough, it doesn't navigate to a different page.

File: packages/data/json/building-types/multi-dwelling/wt-multi-dwelling-9.10.14.json
Comment (nicholaslyon-dt) [reply packages/data/json/building-types/multi-dwelling/wt-multi-dwelling-9.10.14.json:1]: yep! I just have enough test data in there to make sure things work.

**Review by jbook24 — APPROVED** [review]

**Pull request merged**